### PR TITLE
Fetch all attachments when opening a gallery in the Media Library

### PR DESCRIPTION
**Pull request by danielbachhuber**
## Description

Switches `wp.media.query` to use `posts_per_page=-1` and fetch all attachments when opening a gallery in the Media Library.

See https://github.com/WordPress/gutenberg/issues/10873#issuecomment-437234144

## How has this been tested?

1. Create a Gallery Block and click Edit:

image

2. Inspect the Network Tab and verify that `posts_per_page=-1` was used in the request and that the request was successful:

image

### CODE REVIEW

**Comment by adamsilverstein** [comment]
Why are we using unbounded queues vs pagination?

**Comment by joemcgill** [comment]
I just double checked, and can confirm that in the classic editor we are including `posts_per_page=-1` when the `query-attachments` action is called to get data for a gallery. In this case, it seems to make sense, because we are pairing that with a `post__in` value containing an array of all image IDs in the gallery.

1. I don't see anything in this PR that limits this to only apply to galleries, but I think it's a reasonable expectation that we would fetch all of the attachment data needed by any block that is saving a set of attachment IDs.
2. `wp.media.query` actually uses `perPage` as the property instead of `per_page`, which gets mapped to `posts_per_page` [here](https://github.com/WordPress/wordpress-develop/blob/6b366c6620fdd5960cedcdf80955966a715efa82/src/js/media/models/query.js#L186-L200). So good catch that it was wrong. Changing it to `posts_per_page` directly should work fine, however.

File: packages/edit-post/src/hooks/components/media-upload/index.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ const getAttachmentsCollection = ( ids ) => {
 		order: 'ASC',
 		orderby: 'post__in',
 		post__in: ids,
-		per_page: 100,
+		posts_per_page: -1,
```
Comment (joemcgill) [inline packages/edit-post/src/hooks/components/media-upload/index.js]: As noted in my comment, `wp.media.query()` normally uses `perPage` instead of `posts_per_page`, but either should work and I slightly prefer the fact that this is consistent with the WP_Query args that are more familiar.

TL;DR – LGTM.

**Review by joemcgill — APPROVED** [review]
